src/Compilers/CSharp/Portable/Symbols/PublicModel/ErrorTypeSymbol.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+using System.Diagnostics;
+
+namespace Microsoft.CodeAnalysis.CSharp.Symbols.PublicModel
+{
+    internal sealed class ErrorTypeSymbol : NamedTypeSymbol, IErrorTypeSymbol
+    {
+        private readonly Symbols.ErrorTypeSymbol _underlying;
+
+        public ErrorTypeSymbol(Symbols.ErrorTypeSymbol underlying, CodeAnalysis.NullableAnnotation nullableAnnotation)
+            : base(nullableAnnotation)
+        {
+            Debug.Assert(underlying is object);
+            _underlying = underlying;
+        }
+
+        protected override ITypeSymbol WithNullableAnnotation(CodeAnalysis.NullableAnnotation nullableAnnotation)
+        {
+            Debug.Assert(nullableAnnotation != _underlying.DefaultNullableAnnotation);
+            Debug.Assert(nullableAnnotation != this.NullableAnnotation);
+            return new ErrorTypeSymbol(_underlying, nullableAnnotation);
+        }
+
+        internal override CSharp.Symbol UnderlyingSymbol => _underlying;
+        internal override Symbols.NamespaceOrTypeSymbol UnderlyingNamespaceOrTypeSymbol => _underlying;
+        internal override Symbols.TypeSymbol UnderlyingTypeSymbol => _underlying;
+        internal override Symbols.NamedTypeSymbol UnderlyingNamedTypeSymbol => _underlying;
+
+        ImmutableArray<ISymbol> IErrorTypeSymbol.CandidateSymbols => _underlying.CandidateSymbols.SelectAsArray(s => s.GetPublicSymbol());
```

Comment:
>_underlying.CandidateSymbols.SelectAsArray(s => s.GetPublicSymbol()) [](start = 69, length = 68)

Is this something we want to cache?

Reply:
> Is this something we want to cache?

Time will tell. I do not want to cache prematurely. For example, we don't cache members for ```SubstitutedNamedTypeSymbol``` and this API will not be used nearly that often.  

---
In reply to: [338823971](https://github.com/dotnet/roslyn/pull/39498#discussion_r338823971) [](ancestors = 338823971)